cpp/src/Ice/LoggerAdminI.cpp

```diff
@@ -736,11 +737,12 @@ LoggerAdminLoggerI::run()
 
     for(;;)
     {
-        IceUtil::Monitor<IceUtil::Mutex>::Lock lock(_monitor);
-        while(!_destroyed && _jobQueue.empty())
+        unique_lock lock(_mutex);
+        while (!_destroyed && _jobQueue.empty())
         {
-            _monitor.wait();
+            _conditionVariable.wait(lock);
```

Comment:
can we use wait with a callback function here?

Reply:
Yes I fixed this